src/main/java/com/minecolonies/coremod/commands/MinecoloniesCommand.java

```diff
@@ -34,4 +41,41 @@ public MinecoloniesCommand()
     {
         return subCommands;
     }
+
+    /**
+     * Check if the player is able to execute a teleport command again.
+     * @param player
+     * @return
+     */
+    public static boolean canExecuteCommand(@NotNull EntityPlayer player)
+    {
+        if(Configurations.teleportBuffer == 0)
+        {
+            return true;
+        }
+
+        cleanUpList();
+        final boolean canTeleport = !commandExecutions.containsKey(player.getUniqueID());
+
+        if(canTeleport)
+        {
+            commandExecutions.put(player.getUniqueID(), Instant.now());
+        }
+        return canTeleport;
+    }
+
+    /**
+     * Clean up the commandExecutions list according to config details.
+     */
+    private static void cleanUpList()
+    {
+        final Map<UUID, Instant> mapCopy = new HashMap<>(commandExecutions);
+        for(Map.Entry<UUID, Instant> entry : mapCopy.entrySet())
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR') Local variable 'entry' could be declared final [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=pmd%3ALocalVariableCouldBeFinal)